packages/popover/src/usePopover.ts

```diff
@@ -1,38 +1,45 @@
-import { computed, watch } from 'vue'
+import type { SetupContext } from 'vue'
+import { onMounted, ref, watch, watchEffect } from 'vue'
 import { isString } from '@element-plus/utils/util'
+import type { IPopperOptions } from '@element-plus/popper'
 import { usePopper } from '@element-plus/popper'
 import PopupManager from '@element-plus/utils/popup-manager'
+import { EmitType } from '@element-plus/popper/src/use-popper'
 
-import type { IPopperOptions } from '@element-plus/popper'
-import type { SetupContext } from 'vue'
-
-export interface IUsePopover extends IPopperOptions{
+export interface IUsePopover extends IPopperOptions {
   width: number | string
 }
 
 export const SHOW_EVENT = 'show'
 export const HIDE_EVENT = 'hide'
 
 export default function usePopover(props: IUsePopover, ctx: SetupContext<string[]>) {
-  const popperStyle = computed(() => {
+  const popperStyle = ref({})
 
+  const updatePopperStyle = () => {
     let _width: string
 
     if (isString(props.width)) {
       _width = props.width as string
     } else {
       _width = props.width + 'px'
     }
-
-    return {
+    popperStyle.value = {
       width: _width,
       zIndex: PopupManager.nextZIndex(),
     }
-  })
+  }
 
-  const popperProps = usePopper(props, ctx)
+  const popperProps = usePopper(props, ctx as SetupContext<EmitType[]>)
+
+  watchEffect(() => {
+    updatePopperStyle()
+  })
```

Comment:
I think we only need to update the `zIndex` when `visibility` is set to true.

Reply:
Like the other components do, such as `Dialog` `Drawer`

Reply:
when width changes, the unit test won't pass.

Reply:
I mean `width` and `zIndex` should be separated, because they are not updated via the same condition, so I think it's better to split it up.